core/src/main/java/org/axonframework/eventhandling/EventProcessor.java

```diff
@@ -59,6 +59,20 @@
      */
     Registration registerInterceptor(MessageHandlerInterceptor<? super EventMessage<?>> interceptor);
 
+    /**
+     * Sets the event handler invoker to do the actual invocation of handlers.
+     *
+     * @param eventHandlerInvoker the event handler invoker
+     */
+    void setEventHandlerInvoker(EventHandlerInvoker eventHandlerInvoker);
```

Comment:
I would try to avoid introducing mutability in event processors. Instead, the configuration should maintain a collection of processing groups (consisting of a number of invokers). That will allow processors to be created after all the processing groups have been defined.

Probably, the SagaConfiguration should have an extra method (instead of assigning to a (new) subscribing or tracking processor) to assign a Saga to an already defined processor in the Configuration's EventHandlingConfiguration. You can find that one by iterating over the modules, and doing an instanceOf check. If there are multiple `EventHandlingConfiguration`s, throw an exception (always better than guessing)

Reply:
Kinda like this idea!